### PR TITLE
Added support for double in batch norm

### PR DESCRIPTION
**Pull request by prabhat00155**
**Description**:
Added double as supported datatype in batch norm op.

**Motivation and Context**
Fixes https://github.com/microsoft/onnxruntime/issues/2874

### CODE REVIEW

File: onnxruntime/core/providers/cpu/nn/batch_norm.cc [hunk 1]
```diff
@@ -37,11 +37,16 @@ ONNX_CPU_OPERATOR_KERNEL(
     BatchNormalization,
     9,
     KernelDefBuilder()
-        .TypeConstraint("X", DataTypeImpl::GetTensorType<float>())
-        .TypeConstraint("scale", DataTypeImpl::GetTensorType<float>())
-        .TypeConstraint("B", DataTypeImpl::GetTensorType<float>())
-        .TypeConstraint("mean", DataTypeImpl::GetTensorType<float>())
-        .TypeConstraint("var", DataTypeImpl::GetTensorType<float>()),
+        .TypeConstraint("X", {DataTypeImpl::GetTensorType<float>(),
```
Comment (snnn) [inline onnxruntime/core/providers/cpu/nn/batch_norm.cc]: So X could be double while scale is float?

**Comment by snnn** [comment]
You should set T to float or double, not the individual parameters.

https://github.com/onnx/onnx/blob/master/docs/Operators.md#type-constraints-12

File: onnxruntime/core/providers/cpu/nn/batch_norm.cc (hunk 1 above)
Comment (prabhat00155) [reply onnxruntime/core/providers/cpu/nn/batch_norm.cc]: I have set T to float or double, could you take a look now?

**Comment by snnn** [comment]
There isn't any implementation code for it? I only saw  BatchNorm<float>

**Comment by jywu-msft** [comment]
> There isn't any implementation code for it? I only saw BatchNorm<float>

please also add a unit test.
